pkg/cmd/util/route.go

```diff
@@ -54,10 +57,16 @@ func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName,
 		},
 	}
 
-	// If the service has multiple ports and the user didn't specify --port,
-	// then default the route port to a service port name.
-	if len(port.Name) > 0 && len(portString) == 0 {
-		route.Spec.Port = resolveRoutePort(port.Name)
+	// When route.Spec.Port is not set, the generator will pick a service port.
+
+	// If the user didn't specify --port, and either the service has a port.Name
+	// or forcePort is set, set route.Spec.Port
+	if (len(port.Name) > 0 || forcePort) && len(portString) == 0 {
+		if forcePort {
+			route.Spec.Port = &api.RoutePort{TargetPort: svc.Spec.Ports[0].TargetPort}
```

Comment:
Sorry, I was advocating that we just do:

route.Spec.Port = resolveRoutePort(svc.Spec.Ports[0].TargetPort)

Or does that not work for some reason?